lib/transbank/sdk/onepay/models/shopping_cart.rb

```diff
@@ -1,68 +1,80 @@
-require 'transbank/sdk/onepay/utils/jsonify'
-
 module Transbank
   module Onepay
-    # TODO: YARD this class
+    # Represents a Shopping Cart, which contains [Item]s that the user wants to buy
     class ShoppingCart
-      include Utils::JSONifier
-      # Represents a Shopping Cart, which contains [Item]s that the user wants to buy
-
-
+      include Utils::JSONUtils
       # @return [Integer] The amount in CLP of the [Item]s included in the [ShoppingCart]
       attr_reader :total
 
-      def initialize
+      # @param items [Array, nil] an array of Hashes that can be converted to [Item]
+      # if nil, an empty shopping cart is created
+      def initialize(items = [])
         # The amount in CLP of the [Item]s included in the [ShoppingCart]
         @total = 0
         # An [Array<Item>] with the [ShoppingCart] contents
         @items = []
+        return if items.nil? || items.empty?
+
+        items.each do |it|
+          it = transform_hash_keys it
+          item = Item.new it
+          self << item
+        end
       end
 
       # @param item [Item] an instance of [Item]
       # @return [boolean] return true if item is successfully added
       def add(item)
-        new_total = calculate_total(item)
+        new_total = total + item.total
+        if new_total < 0
+          raise Errors::ShoppingCartError, "New total amount cannot be less than zero."
+        end
         @items << item
         @total = new_total
         true
       end
 
-      # Call #add(item) on self, see docs for #add
+      # Alias for #add
       def << item
         add item
       end
 
+      # Remove an [Item] from self
+      # @return [boolean] return true if the item is successfully removed
       def remove(item)
-        new_total = calculate_total(item)
+        new_total = total - item.total
+        if new_total < 0
+          raise Errors::ShoppingCartError, "New total amount cannot be less than zero."
+        end
         first_instance_of_item = @items.index(item)
```

Comment:
Why you don't just use the `delete` method? Sounds like your `==` implementation is making things harder here rather than easier

Reply:
The == method on item is so you can use remove the item like it was done here, otherwise you have to have a reference to the exact Item instance to remove it (or do the comparison between item properties here)

Reply:
No. Given your `==` implementation you can just use `delete` and don't need to complicate the code reinventing it.

Reply:
Just tested it, you cant use #delete since it will delete ALL items that have the same attributes